### PR TITLE
support arbitrary dtype in protobuf

### LINKED ISSUE #1262: Support Arbitrary Data type in protobuf
Currently, we only support data type float32 in protobuf. 

I am planning to refactor the evaluation process to fix issue #1082. To support metrics inherited from `tf.Metric`, worker need to transport labels and model outputs to master. We can't suppose that labels and model outputs are of type `float32`. So we need to support arbitrary data type in protobuf.

### PR DESCRIPTION
**Pull request by mhaoli**
Fixes #1262.

### CODE REVIEW

File: elasticdl/proto/elasticdl.proto [hunk 1]
```diff
@@ -53,6 +53,9 @@ message Tensor {
     // Indices will be tf.IndexedSlices.indices if the tensor is in the form
     // of tf.IndexedSlices. Ohterwise indices will be None.
     repeated int32 indices = 3;
+
+    // Dtype of the tensor, e.g. "int64", "float32".
+    string dtype = 4;
```
Comment (QiJune) [inline elasticdl/proto/elasticdl.proto]: Data type should be an enum type, string is too flexible.

Comment (skydoorkai) [reply elasticdl/proto/elasticdl.proto]: `enum` is clear. `string` can support arbitrary types (supported by Numpy) and simple to code.

Comment (mhaoli) [reply elasticdl/proto/elasticdl.proto]: Use enum type instead.

**Review by QiJune — APPROVED** [review]
Excellent

File: elasticdl/python/tests/ndarray_test.py [hunk 2]
```diff
@@ -11,10 +11,6 @@
 
 class ConverterTest(unittest.TestCase):
     def test_ndarray_to_tensor(self):
-        # Wrong type, should raise
```
Comment (terrytangyuan) [inline elasticdl/python/tests/ndarray_test.py:14]: Should we use this space to test `DT_INVALID` instead of removing this test?

Comment (mhaoli) [reply elasticdl/python/tests/ndarray_test.py:14]: Good point. Added.

**Review by terrytangyuan — APPROVED** [review]

**Pull request merged**